source/js/searcher.js

```diff
@@ -0,0 +1,771 @@
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in all
+//  copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+//  SOFTWARE.
+//  
+//
+//  Searcher API
+//
+//  Author: Radovan Jakubcik
+//
+//  Summary:
+//  Searcher API enables highlighting of the current website content, live search throughout the included websites of the project,
+//  handling of GET request searching in the so called search website. (The website used for sole purpose of searching/displaying results of search)
+//  Enables the users to search for content in real-time or after pressing the 'return' key. (Enter)
+//  Quick and fluent searching of the content is done by loading the content of the website, afterwards inserting the content into the sessionStorage object
+//  using the available languages as the keys in the sessionStorage object. The content is saved under the language and is an object with the following properties
+//  path, content and because the sessionStorage only allows for the storage of strings. The object is afterwards converted into JSON format and stored.
+//    
+//
+//  Requirements:
+//  The script will only work if the url scheme used when loading the website is HTTP or HTTPS for CORS request, otherwise it will automatically be disabled.
+//
+//  Web browser (version) requirements for Session Storage object:
+//  Chrome - 4.0
+//  Microsoft Edge - 8.0
+//  Firefox - 3.5
+//  Safari - 4.0
+//  Opera - 11.5
+//  Safari (iOS) - 3.2
+//  Android Browser - 90
+//  Opera Mobile - 12
+//  Chrome (Android) - 90
+//  Firefox (Android) - 87
+//  Samsung Internet - 4
+//
+//  Script uses the file provided in by the constant defined (is editable):
+//  
+//  C.FILE_PATH
+//
+//  This file has to be created before using this script.
+//
+//  Default file is called: htmlfiles.json
+//  The structure of the file should be as follows (JSON format):
+//  [[The list of websites],[The list of languages]]
+//  [["/example/index.html","/index.html","/example2/example/index.html"],["en","sk"]]
+
+const CURRENT_URL_SCHEME = location.protocol.split(":")[0].toUpperCase();
```

Comment:
maybe add mini explanation for extra clarity

Reply:
done.
+ Added extra explanation on how to load the website correctly for the searcher API to work. Into the head section, before the requirements paragraph.